pkg/operator/ceph/object/user/controller.go

```diff
@@ -309,8 +309,17 @@ func (r *ReconcileObjectStoreUser) initializeObjectStoreContext(u *cephv1.CephOb
 	if err != nil {
 		return errors.Wrapf(err, "Multisite failed to set on object context for object store user")
 	}
+
 	r.objContext = objContext
-	r.objContext.Endpoint = store.Status.Info["endpoint"]
+	if store.Status.Info["endpoint"] == "" && store.Status.Info["secureEndpoint"] == "" {
```

Comment:
What about this approach?
```go
r.objContext.Endpoint = store.Status.Info["endpoint"]
if store.Status.Info["secureEndpoint"] != "" {
  r.objContext.Endpoint = store.Status.Info["secureEndpoint"]
}
if r.objContext.Endpoint == "" {
  return errors.Errorf("endpoint is missing in the status Info")
}
```

Reply:
@travisn :
At least from testing of Red Hat QA, the `endpoint` was not filled properly even though for `cephobjectstoreuser`, even though it was part of Status.Info. So I am suspecting a race between filling up `status.Info` during ceph object store creation and  `cephobjectstoreuser` creation happens simultaneously.

Status.Info by `updateStatus()` in `reconcile()` after calling `reconcileCreateObjectStore()`. If `cephobjectstoreuser`() come after the `reconcileCreateObjectStore` and before `updateStatus()` then value of Endpoint become nil. So to avoid that idea is fail creation request and retry it again. We can also check `Status.Phase` instead of string null check of `endpoint` and `secureendpoint`

Reply:
I don't see how this change could fix a race condition. It's just setting the `objContext.Endpoint` to the `secureEndpoint` if the `endpoint` is not set.

btw my code suggestion above was just to restructure the code, but should be functionally equivalent to your code change.

Reply:
@travisn : I agree it won't fix any race. But it errors out if both `endpoint` and `secureEndpoint` is null which can happen only if a race happens. Or else we can make `objectcontext.Endpoint` not rely on `Status.Info` fill by itself using `buildStatusInfo`

Reply:
@thotz What do you think about the change in code structure I suggested? It just reads simpler to me, even if functionally the same as your change.

Reply:
Sure